File: 08_zero_sum_subarrays.c
#include <stdio.h>

int zeroSumSubarrays(int arr[], int n) {
    int count = 0;
    for (int i = 0; i < n; i++) {
        int sum = 0;
        for (int j = i; j < n; j++) {
            sum += arr[j];
            if (sum == 0) count++;
        }
    }
    return count;
}

int main() {
    int arr[] = {6, 3, -1, -3, 4, -2, 2, 4, 6, -12, -7};
    int n = 11;
    printf("%d\n", zeroSumSubarrays(arr, n));
    return 0;
}

File: 08_zero_sum_subarrays.py
def zero_sum_subarrays(arr):
    prefix_sum = 0
    freq = {0: 1}
    count = 0
    for num in arr:
        prefix_sum += num
        count += freq.get(prefix_sum, 0)
        freq[prefix_sum] = freq.get(prefix_sum, 0) + 1
    return count

if __name__ == "__main__":
    arr = [6, 3, -1, -3, 4, -2, 2, 4, 6, -12, -7]
    print(zero_sum_subarrays(arr))
